instance/tests/api/test_openedx_appserver.py

```diff
@@ -145,7 +145,7 @@ def test_get_details(self):
         self.assertIn(('public_ip', None), server_data)
         # The API call will try to start the server, which will fail, since we're
         # not actually talking to an OpenStack instance when unit tests are running
-        self.assertIn(('status', 'failed'), server_data)
+        self.assertIn(('status', 'unknown'), server_data)
```

Comment:
@bradenmacdonald  Ah yes, thanks for catching that.  I couldn't figure out what had changed, and so thought it was something with OpenStack, or nova client, or elsewhere.  My apologies!

I pinned it down to this exception:
```
Traceback (most recent call last):
  File "/home/vagrant/opencraft/instance/models/server.py", line 330, in update_status
    os_server = self.os_server
  File "/home/vagrant/opencraft/instance/models/server.py", line 285, in os_server
    self.start()
  File "/home/vagrant/opencraft/instance/models/utils.py", line 295, in wrapped_method
    return method(resource, *args, **kwargs)
  File "/home/vagrant/opencraft/instance/models/server.py", line 372, in start
    key_name=settings.OPENSTACK_SANDBOX_SSH_KEYNAME,
  File "/home/vagrant/opencraft/instance/openstack.py", line 74, in create_server
    flavor = nova.flavors.find(**flavor_selector)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/base.py", line 407, in find
    matches = self.findall(**kwargs)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/base.py", line 463, in findall
    listing = self.list(**list_kwargs)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/v2/flavors.py", line 132, in list
    return self._list("/flavors%s%s" % (detail, query_string), "flavors")
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/base.py", line 249, in _list
    resp, body = self.api.client.get(url)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/client.py", line 480, in get
    return self._cs_request(url, 'GET', **kwargs)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/client.py", line 436, in _cs_request
    self.authenticate()
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/client.py", line 612, in authenticate
    auth_url = self._v1_auth(auth_url)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/client.py", line 648, in _v1_auth
    resp, body = self._time_request(url, 'GET', headers=headers)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/client.py", line 431, in _time_request
    resp, body = self.request(url, method, **kwargs)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/novaclient/client.py", line 396, in request
    **kwargs)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/requests/sessions.py", line 596, in send
    r = adapter.send(request, **kwargs)
  File "/home/vagrant/.virtualenvs/opencraft/lib/python3.5/site-packages/requests/adapters.py", line 482, in send
    raise RetryError(e, request=request)
requests.exceptions.RetryError: HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: /auth/url (Caused by ResponseError('too many 500 error responses',))
```

In `.env.test`, `OPENSTACK_AUTH_URL='http://example.com/auth/url'`.  The failing tests are hitting that URL, and balking at the 500 error it returns.  I don't know if `example.com` used to placidly return status code 200, but now it returns 500, so we have to mock it.

e079ae2 reverts my naughty test data changes, and 67cb886 mocks the `OPENSTACK_AUTH_URL` response.  The tests are succeeding on my local devstack; hopefully CircleCI will agree.

Is it ok to include this fix in this PR, or do I need to split it into a separate one?